app/services/supplementary-billing/abstraction-billing-period.service.js

```diff
@@ -133,56 +126,86 @@ function _abstractionPeriods (billingPeriod, chargePurpose) {
     endDate: _subtractOneYear(firstPeriod.endDate)
   }
 
-  return [previousPeriod, firstPeriod]
+  const abstractionPeriods = []
+  if (_isPeriodValid(previousPeriod, billingPeriod)) {
+    abstractionPeriods.push(previousPeriod)
+  }
+
+  if (_isPeriodValid(firstPeriod, billingPeriod)) {
+    abstractionPeriods.push(firstPeriod)
+  }
+
+  return abstractionPeriods
 }
 
 /**
- * Calculates the 'billable days' for each abstraction billing period
+ * Calculates the 'billable days' for a billable abstraction period
  *
- * @param {Object[]} abstractionPeriods An array of abstraction billing periods
+ * @param {Object} abstractionPeriod An abstraction billing period
  *
- * @returns {Object[]} The array abstraction periods each with a new `billableDays` property
- * @returns {Date} periods[].startDate
- * @returns {Date} periods[].endDate
- * @returns {boolean} periods[].consider
- * @returns {number} periods[].billableDays
+ * @returns {Object} The abstraction period with a new `billableDays` property
  */
-function _calculateBillableDays (abstractionPeriods) {
-  let difference
-  let billableDays
-  for (const abstractionPeriod of abstractionPeriods) {
-    difference = abstractionPeriod.endDate.getTime() - abstractionPeriod.startDate.getTime() // difference in msecs
-    billableDays = Math.ceil(difference / (1000 * 3600 * 24)) + 1 // (1000 msecs * (60 secs * 60 mins) * 24 hrs)
+function _calculateBillableDays (abstractionPeriod) {
+  const DAY_IN_MILLISECONDS = (24 * 60 * 60 * 1000) // (24 hrs * 60 mins * 60 secs * 1000 msecs)
+  if (abstractionPeriod.billableStartDate) {
+    const difference = abstractionPeriod.billableEndDate.getTime() - abstractionPeriod.billableStartDate.getTime() // difference in msecs
+    const billableDays = Math.ceil(difference / DAY_IN_MILLISECONDS) + 1
     abstractionPeriod.billableDays = billableDays
   }
 }
 
 /**
- * Adds the `consider` flag to each abstraction billing period as to whether it should be included in billing
+ * Calculates the 'billable period' for an abstraction billing period
+ *
+ * @param {Object} abstractionPeriod An abstraction billing period
+ * @param {Object} billingPeriod Object that has a `startDate` and `endDate` that defines the billing period
+ *
+ * @returns {Object} The abstraction period with new `billableStartDate` & `billableEndDate` properties
+ */
+function _calculateBillablePeriods (abstractionPeriod, billingPeriod) {
```

Comment:
Now we're only looking at an individual period, I think it would be clearer to tweak the name.

```suggestion
function _calculateBillablePeriod (abstractionPeriod, billingPeriod) {
```